### ISSUE TITLE
[BUG] behavior of "stretch_both" stops working for `row()` between bokeh 2.3.0 and 2.3.1

### ISSUE THREAD

**Issue by stevendbrown** [bokeh/bokeh#11227]
Thank you for this amazingly time-saving package! I apologize in advance for any deficiencies in this bug report and hope that I may have the opportunity to rectify them.

# Software versions
Browsers tested (on macOS test environment):
- Brave 1.22.70
- Chrome 90.0.4430.93

Test environment 1 (where minimal working example was built):
- macOS 10.15.7
- Python 3.9.1
- a new virtual environment not used for anything else, only pip command being `pip3 install bokeh==2.3.x` where `x` is 0 or 1.

Test environment 2 (where upgrading to 2.3.1 broke the layout in a larger application):
- Linux CentOS 7.5.1804
- Python 3.8.6

Test environment 1 package versions with bokeh 2.3.1 (`pip3 install bokeh==2.3.1`):
`pip3 freeze`:
```
bokeh==2.3.1
Jinja2==2.11.3
MarkupSafe==1.1.1
numpy==1.20.2
packaging==20.9
Pillow==8.2.0
pyparsing==2.4.7
python-dateutil==2.8.1
PyYAML==5.4.1
six==1.15.0
tornado==6.1
typing-extensions==3.7.4.3
```

Test environment 1 package version with bokeh 2.3.0 (`pip3 install bokeh==2.3.0`):
`pip3 freeze`:
```
bokeh==2.3.0
Jinja2==2.11.3
MarkupSafe==1.1.1
numpy==1.20.2
packaging==20.9
Pillow==8.2.0
pyparsing==2.4.7
python-dateutil==2.8.1
PyYAML==5.4.1
six==1.15.0
tornado==6.1
typing-extensions==3.7.4.3
```

# Expected behavior

The button and plot in the row will occupy all available vertical browser space.

# Observed behavior

- With bokeh 2.3.0, the button and plot occupy all available vertical browser space and are responsive to vertical resizing.
- With bokeh 2.3.1, the button and plot do not occupy all available vertical browser space and are not responsive to vertical resizing.

# Code sample to reproduce

```
from bokeh.layouts import row
from bokeh.models import ColumnDataSource, Button
from bokeh.plotting import figure
from bokeh.io import show

source = ColumnDataSource()
source.data = {
    'val_a': [0, 1],
    'val_b': [0, 1],
}

p = figure(
    sizing_mode="stretch_both",
)

p.circle(
    x='val_a',
    y='val_b',
    source=source,
)

abutton = Button(
    label="a button",
    sizing_mode="stretch_both"
)
    
layout = row(abutton, p, sizing_mode='stretch_both')

show(layout)
```

# Stack traceback and/or browser JavaScript console output

No exceptions occur, and no JavaScript errors occur in browser.

# Screenshots or screencasts of the bug in action

With bokeh 2.3.0:
<img width="1469" alt="with_bokeh_2 3 0" src="https://user-images.githubusercontent.com/7763249/116617647-f0c61f00-a8f2-11eb-909d-520aba7d6a13.png">

With bokeh 2.3.1:
<img width="1469" alt="with_bokeh_2 3 1" src="https://user-images.githubusercontent.com/7763249/116617672-f7ed2d00-a8f2-11eb-9aac-91e1b1ef79c9.png">

**Comment by stevendbrown** [comment]
Just seeing now that this might be related to #11188 and/or #11161. Derp.

**Comment by tcmetzger** [comment]
@stevendbrown Sorry for the late reply - the bug you experienced has been fixed. I tested your code example with Windows and Ubuntu using the upcoming 2.4 release and everything works as expected. 2.4 is expected to be released in the next few weeks!

**Comment by mattpap** [comment]
This was published in 2.3.3.

**Comment by stevendbrown** [comment]
Thank you for the note @tcmetzger !
